protocols/relay/src/behaviour/handler.rs

```diff
@@ -502,39 +470,21 @@ impl Handler {
                 self.pending_error = Some(StreamUpgradeError::Io(e));
                 return;
             }
-            StreamUpgradeError::Apply(error) => match error {
-                outbound_stop::UpgradeError::Fatal(error) => {
-                    self.pending_error = Some(StreamUpgradeError::Apply(Either::Right(error)));
-                    return;
-                }
-                outbound_stop::UpgradeError::CircuitFailed(error) => {
-                    let status = match error {
-                        outbound_stop::CircuitFailedReason::ResourceLimitExceeded => {
-                            proto::Status::RESOURCE_LIMIT_EXCEEDED
-                        }
-                        outbound_stop::CircuitFailedReason::PermissionDenied => {
-                            proto::Status::PERMISSION_DENIED
-                        }
-                    };
-                    (StreamUpgradeError::Apply(error), status)
-                }
-            },
+            StreamUpgradeError::Apply(_) => unreachable!("Should not emit handle errors"),
```

Comment:
The handler error should be a `Void` so you can do:

```suggestion
            StreamUpgradeError::Apply(v) => void::unreachable(v),
```